src/Orleans/Runtime/GrainReference.cs

```diff
@@ -20,6 +21,8 @@ public class GrainReference : IAddressable, IEquatable<GrainReference>, ISeriali
         [NonSerialized]
         private static readonly TraceLogger logger = TraceLogger.GetLogger("GrainReference", TraceLogger.LoggerType.Runtime);
 
+        private static readonly CancellationTokenManager cancellationTokenManager = new CancellationTokenManager();
```

Comment:
I am worried about this static field in the GR. No real need for it. Lets remove the whole calss CancellationTokenManager , and out its only method as a static method in GR.